doc/source/building/index.rst

```diff
@@ -0,0 +1,433 @@
+
+      .. tab-item:: macOS
+        :sync: macos
+
+        ::
+
+          python -m venv venv
+          source venv/bin/activate
+
+      .. tab-item:: Windows
+        :sync: windows
+
+        ::
+
+          python -m venv venv
+          .\venv\Scripts\activate
+
+    Then install the Python-level dependencies from PyPI with::
+
+       python -m pip install -r all_requirements.txt
+
+To build NumPy in an activated development environment, run::
+
+    spin build
+
+This will install NumPy inside the repository (by default in a
+``build-install`` directory). You can then run tests (``spin test``),
+drop into IPython (``spin ipython``), or take other development steps
+like build the html documentation or running benchmarks. The ``spin``
+interface is self-documenting, so please see ``spin --help`` and
+``spin <subcommand> --help`` for detailed guidance.
+
+
+.. admonition:: IDE support & editable installs
+
+    While the ``spin`` interface is our recommended way of working on NumPy,
+    it has one limitation: because of the custom install location, NumPy
+    installed using ``spin`` will not be recognized automatically within an
+    IDE (e.g., for running a script via a "run" button, or setting breakpoints
+    visually). This will work better with an *in-place build* (or "editable
+    install").
+
+    Editable installs are supported. It is important to understand that **you
+    may use either an editable install or ``spin`` in a given repository clone,
+    but not both**. If you use editable installs, you have to use ``pytest``
+    and other development tools directly instead of using ``spin``.
+
+    To use an editable install, ensure you start from a clean repository (run
+    ``git clean -xdf`` if you've built with ``spin`` before) and have all
+    dependencies set up correctly as described higher up on this page. Then
+    do::
+
+        # Note: the --no-build-isolation is important! meson-python will
+        # auto-rebuild each time NumPy is imported by the Python interpreter.
+        pip install -e . --no-build-isolation
+
+        # To run the tests for, e.g., the `numpy.linalg` module:
+        pytest numpy/linalg
+
+    When making changes to NumPy code, including to compiled code, there is no
+    need to manually rebuild or reinstall. When you run ``git clean -xdf``,
+    which removes the built extension modules, remember to also uninstall NumPy
+    with ``pip uninstall numpy``.
+
+    See the meson-python_ documentation on editable installs for more details
+    on how things work under the hood.
+
+    .. warning::
+
+        Note that editable installs are fundamentally incomplete installs.
+        Their only guarantee is that ``import numpy`` works - so they are
+        suitable for working on NumPy itself, and for working on pure Python
+        packages that depend on NumPy. Headers, entrypoints, and other such
+        things may not be available from an editable install.
```

Comment:
And maybe add a cross-reference for this note to link to.

Reply:
cross-reference and note added